### PR TITLE
python3 compability of urlfetch

### PR DESCRIPTION
**Pull request by peuter**
This PR fixes some errors (bytes vs. str and type conversion with long) when using ldap.schema.urlfetch

### CODE REVIEW

**Comment by encukou** [comment]
Thanks for the patch!
Could you also add a test for this?

**Comment by peuter** [comment]
Well the test would be to run that code in python3. No idea how to write a test for that. As the changes are    quite small and obvious (long type does not exist in python3 anymore and AFAIR `search_subschemasubentry_s` returns a bytes string, so you have to decode it to get an `str`) I do not see the requirement for a test here.

**Comment by cajus** [comment]
Can't this be merged? It's an annoying bug that leads to "can't use upstream when doing stuff with schema".

File: Lib/ldap/schema/subentry.py [hunk 1]
```diff
@@ -460,7 +460,7 @@ def urlfetch(uri,trace_level=0,bytes_mode=None):
     l=ldap.initialize(ldap_url.initializeUrl(),trace_level,bytes_mode=bytes_mode)
     l.protocol_version = ldap.VERSION3
     l.simple_bind_s(ldap_url.who or '', ldap_url.cred or '')
-    subschemasubentry_dn = l.search_subschemasubentry_s(ldap_url.dn)
+    subschemasubentry_dn = l.search_subschemasubentry_s(ldap_url.dn).decode('utf-8')
     if subschemasubentry_dn is None:
       s_temp = None
```
Comment (encukou) [inline Lib/ldap/schema/subentry.py]: This doesn't make sense to me. If `subschemasubentry_dn` can be `None`, then `decode` can't be used on this. Or is the `if subschemasubentry_dn is None:` check redundant?

**Comment by peuter** [comment]
You're right, I changed that and added an additional check for bytes type before decoding

**Comment by encukou** [comment]
This looks good to me, and #81 will provide some testing.

Could I get another set of eyes on this (preferrably @rbarrois)?

File: Lib/ldap/schema/subentry.py [hunk 2]
```diff
@@ -464,6 +464,8 @@ def urlfetch(uri,trace_level=0,bytes_mode=None):
...
     else:
+      if isinstance(subschemasubentry_dn, bytes):
+        subschemasubentry_dn = subschemasubentry_dn.decode('utf-8')
```
Comment (rbarrois) [inline Lib/ldap/schema/subentry.py]: Hmm, I think you just found a nasty bug.

According to [RFC4512](https://tools.ietf.org/html/rfc4512#section-4.2), a subschemaSubentry is always a DN-type entry; which (by specification) *must* be UTF-8 encoded.

This means that this decoding bit should actually happen around the return value of [``search_subschemasubentry_s``](https://github.com/pyldap/pyldap/blob/master/Lib/ldap/ldapobject.py#L864).

Would you mind moving this part there? We should be all good from there :)

**Review by rbarrois — COMMENTED** [review]
Hi! Thanks for the input and for following up with this.

I think you've found a bug in our py3 porting: we should handle some decoding lower in the stack...

**Comment by peuter** [comment]
Ok, done.

**Review by encukou — APPROVED** [review]
LGTM

**Pull request merged**